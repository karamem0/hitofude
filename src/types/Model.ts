//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface File {
  id: string,
  baseName?: string,
  fullName?: string,
  createdDate?: Date,
  updatedDate?: Date,
  downloadUrl?: string,
  parentId?: string
}

export interface FileContent {
  content: string,
  editable?: boolean
}

export interface Folder {
  id: string,
  name?: string,
  createdDate?: Date,
  updatedDate?: Date,
  root?: boolean,
  webUrl?: string,
  parentId?: string,
  folders?: Folder[],
  files?: File[]
}

export interface DialogAction {
  type: DialogType,
  payload: unknown
}

export enum DialogType {
  copyFile = 'copyFile',
  createFile = 'createFile',
  createFolder = 'createFolder',
  deleteFile = 'deleteFile',
  deleteFolder = 'deleteFolder',
  renameFile = 'renameFile',
  renameFolder = 'renameFolder'
}

export interface TabMode {
  type?: TabType,
  open?: boolean
}

export enum TabType {
  explorer = 0,
  search = 1,
}
