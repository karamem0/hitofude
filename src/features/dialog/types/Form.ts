//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface FileCopyDialogFormState {
  baseName?: string
}

export interface FileCreateDialogFormState {
  baseName?: string
}

export interface FileRenameDialogFormState {
  id: string,
  baseName?: string
}

export interface FolderCreateDialogFormState {
  name?: string
}

export interface FolderRenameDialogFormState {
  id: string,
  name?: string
}
