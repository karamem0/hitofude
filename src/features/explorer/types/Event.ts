//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import { FileRejection } from 'react-dropzone';

export interface DropEventData {
  acceptedFiles: File[],
  rejectedFiles: FileRejection[]
}
