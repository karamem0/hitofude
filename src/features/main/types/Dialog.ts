//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export enum DialogType {
  copyFile = 'copyFile',
  createFile = 'createFile',
  createFolder = 'createFolder',
  deleteFile = 'deleteFile',
  deleteFolder = 'deleteFolder',
  renameFile = 'renameFile',
  renameFolder = 'renameFolder'
}

export interface DialogAction {
  type: DialogType,
  payload: unknown
}
