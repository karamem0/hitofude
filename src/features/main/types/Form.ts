//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabItemFormState {
  query?: string
}
