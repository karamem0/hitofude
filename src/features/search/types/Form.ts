//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabPanelFormState {
  query?: string
}

export type SearchTabPanelFormField = keyof SearchTabPanelFormState;
