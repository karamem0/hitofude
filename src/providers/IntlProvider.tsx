//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
