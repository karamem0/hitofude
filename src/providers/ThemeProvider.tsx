//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
