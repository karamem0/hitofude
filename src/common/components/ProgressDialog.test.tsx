//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';
import IntlProvider from '../../providers/IntlProvider';
import Presenter from './ProgressDialog.presenter';
import { ProgressType } from '../../types/Model';
import ThemeProvider from '../../providers/ThemeProvider';

it('should create a shapshot when the type parameter is save', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    mountNode: container,
    type: 'save' as ProgressType
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByText('Saving...')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is upload', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    mountNode: container,
    type: 'upload' as ProgressType
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByText('Uploading...')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is undefined', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    mountNode: container,
    type: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
