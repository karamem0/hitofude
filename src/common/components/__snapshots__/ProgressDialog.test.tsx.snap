// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the type is save 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fui-DialogSurface__backdrop r1e18s3l"
  />
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  grid-gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 1rem;
}

.emotion-1 {
  font-size: 24px;
  line-height: calc(24px * 1.25);
}

<div
    aria-labelledby="dialog-title-r2"
    aria-modal="true"
    class="fui-DialogSurface rsmdyd3"
    data-tabster="{"restorer":{"type":0},"modalizer":{"id":"modal-r1","isOthersAccessible":false,"isTrapped":true}}"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="fui-DialogBody r1h3qql9"
    >
      <div
        class="fui-DialogContent r1v5zwsm emotion-0"
      >
        <div
          class="fui-Spinner r82apo5"
          role="progressbar"
        >
          <span
            class="fui-Spinner__spinner rvgcg50 ___1wbe01p_3bwnur0 fxldao9 f1w9dchk f1v3ph3m"
          >
            <span
              class="fui-Spinner__spinnerTail rxov3xa"
            />
          </span>
        </div>
        <span
          class="fui-Text emotion-1 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
        >
          Saving...
        </span>
      </div>
    </div>
  </div>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <span
      hidden=""
    />
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the type is undefined 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr8 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;

exports[`should match the snapshot when the type is upload 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fui-DialogSurface__backdrop r1e18s3l"
  />
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  grid-gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 1rem;
}

.emotion-1 {
  font-size: 24px;
  line-height: calc(24px * 1.25);
}

<div
    aria-labelledby="dialog-title-r6"
    aria-modal="true"
    class="fui-DialogSurface rsmdyd3"
    data-tabster="{"restorer":{"type":0},"modalizer":{"id":"modal-r5","isOthersAccessible":false,"isTrapped":true}}"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="fui-DialogBody r1h3qql9"
    >
      <div
        class="fui-DialogContent r1v5zwsm emotion-0"
      >
        <div
          class="fui-Spinner r82apo5"
          role="progressbar"
        >
          <span
            class="fui-Spinner__spinner rvgcg50 ___1wbe01p_3bwnur0 fxldao9 f1w9dchk f1v3ph3m"
          >
            <span
              class="fui-Spinner__spinnerTail rxov3xa"
            />
          </span>
        </div>
        <span
          class="fui-Text emotion-1 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
        >
          Uploading...
        </span>
      </div>
    </div>
  </div>
  <div
    class="fui-FluentProvider fui-FluentProviderr4 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <span
      hidden=""
    />
  </div>
</DocumentFragment>
`;
